scripts/hooks/nucypher_dkg.py

```diff
@@ -36,191 +24,224 @@
 
-try:
-    eth_provider_uri = sys.argv[1]
-except IndexError:
-    emitter.message("You must provide a eth (staking) provider URI", color="red")
-    sys.exit(-1)
-
-try:
-    coordinator_provider_uri = sys.argv[2]
-except IndexError:
-    emitter.message("You must provide a coordinator network provider URI", color="red")
-    sys.exit(-1)
-
-try:
-    ritual_id = int(sys.argv[3])
-except IndexError:
-    emitter.message(
-        "You must provide a ritual ID; expected <id> or -1 (to initiate a new ritual)",
-        color="red",
-    )
-    sys.exit(-1)
 
-try:
-    signer_uri = sys.argv[4]
-except IndexError:
-    if ritual_id < 0:
-        emitter.message(
-            "You must provide a signer URI for new ritual transactions if not reusing an existing ritual id",
-            color="red",
-        )
-        sys.exit(-1)
-    signer_uri = None
-
-staking_network = "lynx"
-coordinator_network = "mumbai"
-
-coordinator_agent = ContractAgency.get_agent(
-    agent_class=CoordinatorAgent,
-    registry=InMemoryContractRegistry.from_latest_publication(
-        network=coordinator_network
-    ),
-    provider_uri=coordinator_provider_uri,
-)  # type: CoordinatorAgent
-
-staking_network_registry = InMemoryContractRegistry.from_latest_publication(
-    network=staking_network
+@click.command()
+@click.option(
+    "--eth-provider",
+    "eth_provider_uri",
+    help="ETH staking network provider URI",
+    type=click.STRING,
+    required=True,
 )
-application_agent = ContractAgency.get_agent(
-    agent_class=PREApplicationAgent,
-    registry=staking_network_registry,
-    provider_uri=eth_provider_uri,
-)  # type: PREApplicationAgent
-
-
-#
-# Initial Ritual
-#
-if ritual_id < 0:
-    emitter.echo("--------- Initiating Ritual ---------", color="yellow")
-    # create account from keystore file
-    signer = Signer.from_signer_uri(uri=signer_uri)
-    account_address = signer.accounts[0]
-    emitter.echo(
-        f"Using account {account_address} to initiate DKG Ritual", color="green"
-    )
-
-    password = click.prompt(
-        "Enter your keystore password", confirmation_prompt=False, hide_input=True
-    )
-    signer.unlock_account(account=account_address, password=password)
-    transacting_power = TransactingPower(signer=signer, account=account_address)
+@click.option(
+    "--eth-staking-network",
+    "eth_staking_network",
+    help="ETH staking network",
+    type=click.Choice(["tapir", "lynx"]),
+    default="lynx",
+)
+@click.option(
+    "--coordinator-provider",
+    "coordinator_provider_uri",
+    help="Coordinator network provider URI",
+    type=click.STRING,
+    required=True,
+)
+@click.option(
+    "--coordinator-network",
+    "coordinator_network",
+    help="Coordinator network",
+    type=click.Choice(["mumbai"]),
+    default="mumbai",
+)
+@click.option(
+    "--ritual-id",
+    "ritual_id",
+    "-r",
+    help="Ritual ID; existing id or -1 (to initiate a new ritual)",
+    type=click.INT,
+    required=True,
```

Comment:
```suggestion
    default=-1,
```

This is just a suggestion. Could it make sense to have a default value of -1?

From UX perspective, it sounds weird to me introducing -1 value when actually I want a new ritual ID to be generated.

With this suggestion, user can use `--ritual-id` argument when there is an initiated ritual, or don't use this argument if there is not an initiated ritual (and therefore, a new ritual will be initiated).

WDYT?

Reply:
Agree with Manu here!

Reply:
Good point! ✔️